core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -608,8 +608,9 @@ private[spark] class DAGScheduler(
    *
    * @param job The job whose state to cleanup.
    */
-  private def cleanupStateForJobAndIndependentStages(job: ActiveJob): Unit = {
+  private def cleanupStateForJobAndIndependentStages(job: ActiveJob): HashSet[Stage] = {
     val registeredStages = jobIdToStageIds.get(job.jobId)
+    val removedStages = new HashSet[Stage]
```

Comment:
I think you are only returning removed stages that were running so variables should be named removedRunningStages